flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/CoGroupJoinITCase.java

```diff
@@ -324,6 +329,41 @@ public void invoke(String value) throws Exception {
 		Assert.assertEquals(expectedResult, testResults);
 	}
 
+	@Test
+	public void testCoGroupOperatorWithCheckpoint() throws Exception {
```

Comment:
Maybe mention that this is meant to check the snapshotting and the problem that was previously undetected.